### DISCUSSION TITLE
Javascript Update

### DISCUSSION THREAD

**Discussion by bettywolfy** [nesbox/TIC-80#2177] [Ideas]
How about updating the Javascript?

I would like to be able to use "let" in my code and more modern syntaxes with "class" and "[0, 1, 2, 3, 4].map(i => i * 3)"

**Comment by nesbox** [comment]
We already have an issue https://github.com/nesbox/TIC-80/issues/1191

**Reply by bettywolfy** [reply]
2020 💀

**Reply by Vasdranna** [reply]
I'd wish to use ES7. I don't know why this doesn't get enough atention. TIC-80 would be the best Javascript fantasy console out there that way.

**Reply by nesbox** [reply]
We will change Duktape to QuickJS which is ES6

**Reply by Vasdranna** [reply]
Oh! That's really good news! Thank you! Although I've read that QuickJS supports up to ES2020. Will the implementation in TIC80 be limited to ES6 only?

**Reply by nesbox** [reply]
No limitations, it will support all the features that QuickJS supports.

**Reply by Vasdranna** [reply]
That's amazing! Is there an estimated date for that release? Thank you!

**Reply by nesbox** [reply]
Nobody knows the estimations, sorry :)
I'll try to add it when I'll have some free time.
